app/javascript/controllers/rainy_day_controller.js

```diff
@@ -44,63 +110,140 @@ export default class extends Controller {
-
-//   var newUrl = this.buildMountainProjectUrl(longitude, latitude, zoom);
-//   $(this.options.mp.widgetSelector).attr('src', newUrl);
-// }
-
-// RainyDayController.prototype.buildMountainProjectUrl = function (longitude, latitude, zoom) {
-//   return this.options.mp.urlBase + '?loc=fixed&' +
-//       'x=' + longitude + '&' +
-//       'y=' + latitude + '&' +
-//       'z=' + zoom + '&' +
-//       'h=' + this.options.mp.height
-// }
-
+/*********New method for setInitView (NOT CURRENTLY USED) */
+setInitView() {
+const rainyDayListOption = document.querySelector(this.options.areaListElementSelector);
+const activeLocation = document.querySelector(this.options.activeLabelSelector);
+const description = document.querySelector('[data-role="area-description"]');
+const rockType = document.querySelector('[data-role="area-rock-type"]');
+const driveTime = document.querySelector('[data-role="area-drive-time"]')
+
+if (activeLocation.length < 1) {
+  // console.error('No active list element found')
+  throw new Error("No active list element found")
+}
+this.updateMountainProjectWidget(activeLocation)
+}
+
+/*********For rebuilding the area selection ********/
 // RainyDayController.prototype.handleAreaSelection = function (ev) {
 //   var $target = $(ev.target);
 //   var areaId = $target.data('id');
 //   var $selectedItems = $('[data-id="' + $target.data('id') + '"]');
 
 //   // Do nothing if the clicked element is already active
-//   if ($target.hasClass(this.options.activeSelectorClass))
-//   {
-//       return;
-//   }
-
+//   if ($target.hasClass(this.options.activeSelectorClass)) //activeSelectorClass = active{   
+//   return;}
 //   this.fetchArea(areaId)
 //       .then(this.displayArea.bind(this, $selectedItems));
 // }
 
-// RainyDayController.prototype.fetchArea = function (id) {
-//   var url = '/' + this.watchedAreaSlug + '/rainy-day-options/';
-//   return $.get(url + id);
-// }
+/****************** New method for handling area selection (NOT CURRENTlY USED) ************/
+handleAreaSelection(event) {
+  // let target = document.querySelector(event)
+  let areaId = event
+  let element = document.querySelectorAll(`[data-id="${areaId}"]`)
 
-// RainyDayController.prototype.displayArea = function ($newActiveListItem, response) {
-//   if (!response || response.length == 0) {
-//       console.error('Issue connecting to server :(');
-//   }
+  if (this.areaTarget) {
+    return
+  }
+  this.fetchArea(areaId)
+    .then(this.displayArea(() => {
+      element
+    }))
+  }
 
-//   this.swapActiveClass($newActiveListItem);
-//   this.updatePageInformation(response);
-//   this.updateMountainProjectWidget($newActiveListItem);
+/*********Used this function to make an async function for getting url***********/
+    // RainyDayController.prototype.fetchArea = function (id) {
+    //   var url = '/' + this.watchedAreaSlug + '/rainy-day-options/';
+    //   return $.get(url + id);
+    // }
+    fetchArea = async function(event) {
```

Comment:
Let's follow the standard way to define class methods in javascript classes and refactor this to:

```javascript
    async fetchArea(event) {
        # ...
    }
```

Reply:
Done!